src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptionsTests.cs

```diff
@@ -0,0 +1,574 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.AvoidSingleUseOfLocalJsonSerializerOptions,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.AvoidSingleUseOfLocalJsonSerializerOptions,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class AvoidSingleUseOfLocalJsonSerializerOptionsTests
+    {
+        #region Diagnostic Tests
+
+        [Fact]
+        public Task CS_UseNewOptionsAsArgument()
+        {
+            string source = @"
+using System;
+using System.Text.Json;
+
+internal class Program
+{
+    static void Main(string[] args)
+    {
+        string json = JsonSerializer.Serialize(args, new JsonSerializerOptions { AllowTrailingCommas = true });
+        Console.WriteLine(json);
+    }
+}
+";
+            return VerifyCS.VerifyAnalyzerAsync(source,
+                VerifyCS.Diagnostic(AvoidSingleUseOfLocalJsonSerializerOptions.s_Rule)
+                .WithSpan(9, 54, 9, 110));
```

Comment:
NIT: inline markups look better for diagnostic location

Reply:
> inline markups

What's that? Can you share an example?

Reply:
Well, you know the special brackets, like `[| |]` or `{| |}`
https://github.com/dotnet/roslyn-analyzers/blob/3df8c171697a3c76920e68bd5350fb12de69dbda/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKeyTests.cs#L40